girder_worker/plugins/docker/executor.py

```diff
@@ -118,6 +118,26 @@ def _docker_gc(tempdir):
     return subprocess.Popen(args=(script,), env=env)
 
 
+def validate_task_outputs(task_outputs):
+    """
+    This is called prior to fetching inputs to make sure the output specs are
+    valid. Outputs in docker mode can result in side effects, so it's best to
+    make sure the specs are valid prior to fetching.
+    """
+    for name, spec in task_outputs.iteritems():
+        if spec.get('target') == 'filepath':
+            if 'path' not in spec:
```

Comment:
@jeffbaumes can we make `task['input'] and task['outputs']` a dictionary instead of a list. While generating the task spec automatically it will make it computationally faster to find and modify a specific input or output. Otherwise finding an input in the list is an `O(n)` operation.